### PR TITLE
Fix/summary dimension error

### PR DESCRIPTION
**Pull request by puranban**
- Depends on
  - https://github.com/the-deep/server/pull/1370 

## Changes

- Summary form
  - Show error in the dimension tab
  - Breakdown issue search select input
  - Implement new UI to add the issue

## This PR doesn't introduce any:

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] codegen errors
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [x] permission checks
- [x] translations

### CODE REVIEW

File: \ [hunk 1]
```diff
@@ -0,0 +1,152 @@
+import React, { useMemo, useState, useCallback } from 'react';
```
Comment (AdityaKhatri) [inline \]: The page shouldn't exist.

File: app/views/EditAry/AssessmentRegistryForm/SummaryForm/DimensionItem/SubDimensionItem/index.tsx [hunk 2]
```diff
@@ -245,15 +246,16 @@ function SubDimensionItem(props: Props) {
...
                     freeHeight
                 >
-                    <AddIssueModal
-                        type="dimension"
+                    <IssueSearchSelectInput
+                        name="summaryIssue"
                         subDimension={data.subDimension}
-                        onClose={closeModal}
+                        onChange={noOp}
+                        mode="add"
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/SummaryForm/DimensionItem/SubDimensionItem/index.tsx]: Why do we have this mode?

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/AddIssueModal/index.tsx [hunk 3]
```diff
@@ -27,18 +27,26 @@ const CREATE_ASSESSMENT_REGISTRY_SUMMARY_ISSUE = gql`
                 id
                 label
                 subPillar
-                subPillarDisplay
+                subDimension
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/AddIssueModal/index.tsx]: This page is no longer a modal

File: app/views/EditAry/AssessmentRegistryForm/SummaryForm/PillarItem/SubPillarItem/index.tsx [hunk 4]
```diff
@@ -231,16 +231,12 @@ function SubPillarItem(props: Props) {
...
                     onCloseButtonClick={closeModal}
                     freeHeight
                 >
-                    <AddIssueModal
-                        type="pillar"
-                        subPillar={data.subPillar}
-                        onClose={closeModal}
-                    />
+                    <AddSummaryIssue subPillar={data.subPillar} />
                 </Modal>
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/SummaryForm/PillarItem/SubPillarItem/index.tsx]: Move the modal inside AddSummayIssue, call it AddSummaryIssueModal.

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/AddSummaryIssueModal/index.tsx [hunk 5]
```diff
@@ -64,16 +76,26 @@ function AddIssueModal(props: Props) {
...
                         variant: 'success',
                     },
                 );
+                refetch();
                 onClose();
             }
+            if (errors) {
+                const err = errors as ObjectError[];
+                alert.show(
+                    err?.[0]?.messages,
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/AddSummaryIssueModal/index.tsx]: We have a function to extract nonFieldErrors, can you checkout other forms?

File: app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/AddSummaryIssueModal/index.tsx [hunk 6]
```diff
@@ -141,11 +156,11 @@ function AddIssueModal(props: Props) {
                 actions={(
                     <>
                         <Button
-                            name={undefined}
+                            name={parentId}
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/AddSummaryIssueModal/index.tsx]: should be undefined

File: app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+                pending={false}
+                messageShown
+                emptyIcon
+            />
+            {addIssue && (
+                <AddSummaryIssueModal
+                    type="pillar"
+                    subPillar={data.subPillar}
+                    parentId={selected}
+                    onClose={() => setAddIssue(false)}
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx]: create a separate handler for this

File: app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+        }, [],
+    );
+
+    const issueParams = useCallback(
+        (
+            n: string,
+            issuesData: NonNullable<NonNullable<GetSummarySubIssuesQuery['assessmentRegSummaryIssues']>['results']>[number],
+        ) => ({
+            name: n,
+            data: issuesData,
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx]:
```suggestion
            issueKey: string,
            issuesData: NonNullable<NonNullable<GetSummarySubIssuesQuery['assessmentRegSummaryIssues']>['results']>[number],
        ) => ({
            name: issueKey,
            data: issuesData,
```

File: app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+    ), [
+        name,
+        loading,
+        handleAddNewIssue,
+    ]);
+
+    const handleOnExpansionChange = useCallback(
+        (_: boolean, n: string) => {
+            setSelected((oldValue) => (oldValue === n ? undefined : n));
+        }, [],
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx]:
```suggestion
        (_: boolean, issueKey: string) => {
            setSelected((oldValue) => (oldValue === issueKey ? undefined : issueKey));
        }, [],
```

File: app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+            >
+                <IoAddOutline />
+            </QuickActionButton>
+        </div>
+    ), [
+        name,
+        loading,
+        handleAddNewIssue,
+    ]);
+
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx]: Lets put this directly in actions.

File: app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+        {
+            skip: isNotDefined(selected),
+            variables,
+        },
+    );
+
+    const handleAddNewIssue = useCallback(
+        (n: string) => {
+            setAddIssue(true);
+            setSelected(n);
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx]: I don't think we need to save the id separately as we only edit one issue at a time in this component.

File: app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx [hunk 12]
```diff
@@ -0,0 +1,199 @@
...
+    const [selected, setSelected] = useState<string | undefined>();
+    const [addIssue, setAddIssue] = useState<boolean>(false);
+
+    const variables = useMemo(
+        (): GetSummarySubIssuesQueryVariables => ({
+            subPillar: subPillar as AssessmentRegistrySummarySubPillarTypeEnum,
+            subDimension: subDimension as AssessmentRegistrySummarySubDimensionTypeEnum,
+            parent: selected,
+            page: 1,
+            pageSize: 10,
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx]: Add pagination in this list.

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx (hunk 11 above)
Comment (puranban) [reply app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx]: The ID is saved to expand the collapsable container when clicked on the add icon

File: app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/AddSummaryIssueModal/index.tsx [hunk 13]
```diff
@@ -27,27 +28,41 @@ const CREATE_ASSESSMENT_REGISTRY_SUMMARY_ISSUE = gql`
...
 
 type Props = {
     className?: string;
+    parentId?: string;
     onClose: () => void;
+    refetch: () => void;
 } & ({
     type: 'pillar';
-    subPillar: AssessmentRegistrySummarySubPillarTypeEnum;
+    subPillar?: AssessmentRegistrySummarySubPillarTypeEnum | null;
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/AddSummaryIssueModal/index.tsx]: I don't think subPillar and subDimesion should null or `?`.

File: app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx [hunk 14]
```diff
@@ -0,0 +1,239 @@
...
+
+    const [selected, setSelected] = useState<string | undefined>();
+    const [addIssue, setAddIssue] = useState<boolean>(false);
+    const [page, setPage] = useState<number>(1);
+    const [pageSize, setPageSize] = useState<number>(5);
+
+    const variables = useMemo(
+        (): GetSummarySubIssuesQueryVariables => ({
+            subPillar: subPillar as AssessmentRegistrySummarySubPillarTypeEnum,
+            subDimension: subDimension as AssessmentRegistrySummarySubDimensionTypeEnum,
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx]: Cast is not required

File: app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx [hunk 15]
```diff
@@ -0,0 +1,239 @@
...
+        refetch,
+    } = useQuery<GetSummarySubIssuesQuery, GetSummarySubIssuesQueryVariables>(
+        GET_SUMMARY_SUB_ISSUES,
+        {
+            skip: isNotDefined(selected),
+            variables,
+        },
+    );
+
+    const response = removeNull(issuesResponse?.assessmentRegSummaryIssues);
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx]: use useMemo

File: app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx [hunk 16]
```diff
@@ -0,0 +1,239 @@
...
+        (_: boolean, issueKey: string) => {
+            setSelected((oldValue) => (oldValue === issueKey ? undefined : issueKey));
+        },
+        [],
+    );
+
+    const issueParams = useCallback(
+        (
+            issueKey: string,
+            issuesData: NonNullable<NonNullable<GetSummarySubIssuesQuery['assessmentRegSummaryIssues']>['results']>[number],
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx]: Define type outside

File: app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx [hunk 17]
```diff
@@ -0,0 +1,239 @@
...
+            type,
+        }),
+        [
+            subPillar,
+            subDimension,
+            type,
+        ],
+    );
+
+    const handleModalClose = useCallback(() => setAddIssue(false), []);
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/IssueItem/index.tsx]: use useBooleanState to manage boolean state

File: app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/index.tsx [hunk 18]
```diff
@@ -0,0 +1,163 @@
...
+            type,
+        }),
+        [
+            subPillar,
+            subDimension,
+            type,
+        ],
+    );
+
+    const response = removeNull(data?.assessmentRegSummaryIssues);
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/index.tsx]: use useMemo

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

File: app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/AddSummaryIssueModal/index.tsx (hunk 13 above)
Comment (puranban) [reply app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/AddSummaryIssueModal/index.tsx]: This is conditional rendering based upon `type` and `subPillar` or `subDimension` where one must be null or undefined.

Comment (AdityaKhatri) [reply app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/AddSummaryIssueModal/index.tsx]: @puranban , it should be required as the type: pillar | dimension is there to say that subPillar and subDimension are required.

Comment (AdityaKhatri) [reply app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/AddSummaryIssueModal/index.tsx]:
```suggestion
    subPillar: AssessmentRegistrySummarySubPillarTypeEnum;
    subDimension: undefined;
```

File: app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/AddSummaryIssueModal/index.tsx [hunk 19]
```diff
@@ -27,27 +28,41 @@ const CREATE_ASSESSMENT_REGISTRY_SUMMARY_ISSUE = gql`
...
 } | {
     type: 'dimension';
-    subDimension: AssessmentRegistrySummarySubDimensionTypeEnum;
+    subDimension?: AssessmentRegistrySummarySubDimensionTypeEnum | null;
```
Comment (AdityaKhatri) [inline app/views/EditAry/AssessmentRegistryForm/SummaryForm/AddSummaryIssue/AddSummaryIssueModal/index.tsx]:
```suggestion
    subDimension?: AssessmentRegistrySummarySubDimensionTypeEnum;
    subPillar: undefined;
```

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]

**Pull request merged**